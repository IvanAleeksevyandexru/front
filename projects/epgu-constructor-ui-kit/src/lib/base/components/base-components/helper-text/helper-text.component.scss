@import '../../../../../styles/core/index';

.helper-text {
  @include typ-plain-small;
  color: $color-text-helper;

  &-lg {
    @include typ-plain-text;
  }
}
