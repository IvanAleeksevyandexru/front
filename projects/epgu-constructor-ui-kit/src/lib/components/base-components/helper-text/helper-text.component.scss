@import "../../../../styles/core";

.helper-text {
  @include typ-plain-small;
  color: $color-text-helper;

  &-lg {
    @include typ-plain-text;
  }
}
