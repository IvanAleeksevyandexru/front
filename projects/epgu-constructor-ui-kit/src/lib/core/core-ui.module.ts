import { APP_INITIALIZER, NgModule } from '@angular/core';
import { HTTP_INTERCEPTORS, HttpClient } from '@angular/common/http';
import { SmuEventsService } from '@epgu/epgu-lib';
import { CookieService } from 'ngx-cookie-service';

// Сокращать пути до ./components и ./services нельзя, т.к. будет ошибка при `ng build epgu-constructor --prod`
import { CfAppStateService } from './services/cf-app-state/cf-app-state.service';
import { LocalStorageService } from './services/local-storage/local-storage.service';
import { LocationService } from './services/location/location.service';
import { WINDOW_PROVIDERS } from './providers/window.provider';
import { DeviceDetectorService } from './services/device-detector/device-detector.service';
import { EventBusService } from './services/event-bus/event-bus.service';
import { UnsubscribeService } from './services/unsubscribe/unsubscribe.service';
import { ConfigService } from './services/config/config.service';
import { LoggerService } from './services/logger/logger.service';
import { initApp } from './initializers/app.initializer';
import { HttpCancelInterceptor } from './interceptor/http-cancel/http-cancel.interceptor';
import { HttpCancelService } from './interceptor/http-cancel/http-cancel.service';
import { SessionService } from './services/session/session.service';
import { FocusManagerService } from './services/focus-manager/focus-manager.service';
import { DatesToolsService } from './services/dates-tools/dates-tools.service';
import { UtilsService } from './services/utils/utils.service';
import { ModalService } from '../modal/modal.service';

@NgModule({
  providers: [
    ModalService,
    LocalStorageService,
    DeviceDetectorService,
    EventBusService,
    UnsubscribeService,
    CfAppStateService,
    LocationService,
    ConfigService,
    LoggerService,
    HttpCancelService,
    SessionService,
    FocusManagerService,
    DatesToolsService,
    HttpClient,
    UtilsService,
    WINDOW_PROVIDERS,
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCancelInterceptor,
      multi: true,
    },
    {
      provide: APP_INITIALIZER,
      useFactory: initApp,
      deps: [SmuEventsService, CookieService],
      multi: true,
    },
  ],
})
export class CoreUiModule {}
