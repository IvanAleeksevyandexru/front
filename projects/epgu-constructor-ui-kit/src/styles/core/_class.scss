@import "variables";
@import "functions";

.mb {
  &-0 {
    margin-bottom: 0 !important;
  }
  &-4 {
    margin-bottom: 4px !important;
  }

  &-8 {
    margin-bottom: 8px !important;
  }

  &-12 {
    margin-bottom: 12px !important;
  }

  &-14 {
    margin-bottom: 14px !important;
  }

  &-16 {
    margin-bottom: 16px !important;
  }

  &-24 {
    margin-bottom: 24px !important;
  }

  &-32 {
    margin-bottom: 32px !important;
  }
}

.mt {
  &-4 {
    margin-top: 4px !important;
  }

  &-8 {
    margin-top: 8px !important;
  }

  &-12 {
    margin-top: 12px !important;
  }

  &-14 {
    margin-top: 14px !important;
  }

  &-16 {
    margin-top: 16px !important;
  }

  &-24 {
    margin-top: 24px !important;
  }

  &-32 {
    margin-top: 32px !important;
  }
}

@mixin line {
  content: '';
  display: inline-block;
  margin-right: 8px;
  width: 24px;
  height: 4px;
  position: relative;
  bottom: 3px;
}

.yellow-line {
  &::before {
    background-image: escape-svg($yellow-line-icon);
    @include line;
  }
}

.green-line {
  &::before {
    background-image: escape-svg($green-line-icon);
    @include line;
  }
}

.red-line {
  &::before {
    background-image: escape-svg($red-line-icon);
    @include line;
  }
}
.gray-line {
  &::before {
    background-image: escape-svg($gray-line-icon);
    @include line;
  }
}
