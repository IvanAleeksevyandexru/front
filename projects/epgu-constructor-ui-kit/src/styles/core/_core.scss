img {
  max-width: 100%;
}

*,
*:before,
*:after {
  box-sizing: border-box;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  word-wrap: break-word;
}

h1 {
  @include typ-h1;
}

h2 {
  @include typ-h2;
}

h3 {
  @include typ-h3;
}

h4 {
  @include typ-h4;
}

h5 {
  @include typ-h5;
}

h6 {
  @include typ-h6;
}

p {
  @include typ-plain-text;
  &:not(:first-child) {
    margin-top: 24px;
  }
}

small,
label {
  @include typ-plain-small;
}

.bold,
b,
strong {
  font-weight: $fw-bold;
}

.helper-text {
  @include typ-plain-small;
  color: $color-text-helper;
}

.text--center {
  text-align: center;
}

.text--right {
  text-align: right;
}

.text--justify {
  text-align: justify;
}

sub {
  font-size: $font-xs;
  line-height: 16px;
}
a {
  text-decoration: none;
  color: $color-gosblue;
  cursor: pointer;
  outline: none;

  &:hover {
    color: $color-active-item;
  }
}

button {
  font-family: inherit;
}

strike {
  color: $color-text-helper;
}

blockquote {
  background: $color-osso-grigio;
  border: 1px solid $color-pulsante-premuto;
  box-sizing: border-box;
  border-radius: $border-md;
}

.divider {
  background: $color-divider;
  border-top: 1px solid $color-divider;
}

.submit-button {
  @include screen-button-width;
  button {
    @include screen-inner-button-width;
  }
}

/* Utility classes */

.text-color {
  &--white {
    color: $color-white;
  }
  &--gosblue {
    color: $color-gosblue;
  }
  &--gosred {
    color: $color-gosred;
  }
  &--summer-night {
    color: $color-summer-night;
  }
  &--text-helper {
    color: $color-text-helper;
  }
}

.screen-pad-body {
  display: flex;
  @include tablets-down {
    flex-grow: 1;
  }
  & > * {
    display: flex;
    width: 100%;
    flex-direction: column;
  }
}

.block {
  &--with-icon {
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
    margin-top: 16px;
    &:first-child {
      margin-top: 0;
    }
    & > img {
      flex: 0 0 auto;
      max-width: 25px;
      margin-right: 17px;
      margin-bottom: 0;
      @include phones-down {
        margin-top: 4px;
      }
    }
    .icon-wrapper {
      width: 23px;
      flex: 0 0 auto;
      text-align: center;
      display: inline-block;
      margin-right: 17px;
      img {
        max-width: 100%;
        margin-bottom: 0;
      }
    }
  }
  &--content-centered {
    display: flex;
    justify-content: center;
    align-items: center;
  }
  &--description {
    background: $color-azzurro-alice;
    border-radius: 0 4px 4px 4px;
    padding: 16px;
    margin-top: 4px;
    position: relative;

    > *:first-child:not(.block--description__link-close) {
      margin-top: 0;
    }

    &__link-close {
      margin-top: 16px;
      display: block;
    }

    &--close {
      position: absolute;
      top: 16px;
      right: 16px;
      width: 12px;
      height: 12px;
      cursor: pointer;
      z-index: 2;

      &:before,
      &:after {
        content: '';
        position: absolute;
        width: 2px;
        height: 12px;
        left: 5px;
        background: $color-gosblue;
        transform: rotate(45deg);
      }

      &:before {
        transform: rotate(45deg);
      }

      &:after {
        transform: rotate(-45deg);
      }
    }
  }
}

.container {
  width: 100%;
  max-width: 592px;
  margin-right: auto;
  margin-left: auto;
  padding-left: 0;
  padding-right: 0;
  @media (max-width: 623px) {
    padding-left: 16px;
    padding-right: 16px;
  }
}

.screen-container-mt {
  margin-top: $navigation-mb;
}

.counter::after {
  margin-left: 4px;
  counter-increment: section;
  content: counter(section);
}

.hint-text,
p.hint-text {
  font-size: $font-sm;
  line-height: $font-lg;
  margin-top: 4px;
  color: $color-text-helper;
}
