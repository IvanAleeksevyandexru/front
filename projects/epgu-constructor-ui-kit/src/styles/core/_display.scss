@import 'breakpoint';

// Inspired by bootstrap solution. For more usage detail see https://getbootstrap.com/docs/4.0/utilities/display/
// NOTE: We haven't xs

@mixin displays() {
  &-none {
    display: none;
  }

  &-inline {
    display: inline;
  }

  &-inline-block {
    display: inline;
  }

  &-block {
    display: block;
  }

  &-table {
    display: table;
  }

  &-table-cell {
    display: table-cell;
  }

  &-table-row {
    display: table-row;
  }

  &-flex {
    display: flex;
  }

  &-inline-flex {
    display: inline-flex;
  }
}

.d {
  @include displays();

  &-sm {
    @include phones-up {
      @include displays();
    }
  }

  &-md {
    @include tablets-up {
      @include displays();
    }
  }

  &-lg {
    @include laptop-up {
      @include displays();
    }
  }

  &-xl {
    @include desktops-up {
      @include displays();
    }
  }
}
