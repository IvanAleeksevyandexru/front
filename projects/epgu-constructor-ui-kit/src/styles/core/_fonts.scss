@import "variables";

// Font families
$font-lato: 'Lato';

// Font weights
$fw-thin: 100;
$fw-light: 300;
$fw-regular: 400;
$fw-medium: 500;
$fw-semi-bold: 600;
$fw-bold: 700;
$fw-black: 900;

// Font sizes
$font-xs: 12px;
$font-sm: 14px;
$font-dflt: 16px;
$font-md: 18px;
$font-lg: 20px;
$font-xl: 24px;
$font-xxl: 32px;
$font-extra: 36px;

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-BlackItalic.woff') format('woff');
  font-weight: $fw-black;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Light.woff') format('woff');
  font-weight: $fw-light;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Medium.woff') format('woff');
  font-weight: 500;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-HeavyItalic.woff') format('woff');
  font-weight: $fw-black;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-ThinItalic.woff') format('woff');
  font-weight: $fw-thin;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Black.woff') format('woff');
  font-weight: $fw-black;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Heavy.woff') format('woff');
  font-weight: $fw-black;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Regular.woff') format('woff');
  font-weight: $fw-regular;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-BoldItalic.woff') format('woff');
  font-weight: bold;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-SemiboldItalic.woff') format('woff');
  font-weight: 600;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Semibold.woff') format('woff');
  font-weight: 600;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-LightItalic.woff') format('woff');
  font-weight: $fw-light;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Thin.woff') format('woff');
  font-weight: $fw-thin;
  font-style: $fw-regular;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-MediumItalic.woff') format('woff');
  font-weight: 500;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Italic.woff') format('woff');
  font-weight: $fw-regular;
  font-style: italic;
  font-display: auto;
}

@font-face {
  font-family: $font-lato;
  src: url('../../ui-kit-assets/fonts/Lato/Lato-Bold.woff') format('woff');
  font-weight: bold;
  font-style: $fw-regular;
  font-display: auto;
}
