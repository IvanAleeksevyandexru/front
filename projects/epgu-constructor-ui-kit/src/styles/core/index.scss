@import 'palettes';
@import 'variables';
@import 'fonts';
@import 'functions';
@import 'typography';
@import 'breakpoint';
@import 'display';
@import 'grid';
@import 'tables';
@import 'core';
