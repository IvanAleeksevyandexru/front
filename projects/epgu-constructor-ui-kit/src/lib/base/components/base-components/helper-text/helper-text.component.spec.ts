import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { HelperTextComponent } from './helper-text.component';

const helperTextMock = 'Awesome useful helper text.';

@Component({
  template: `<epgu-cf-ui-constructor-helper-text
    >${helperTextMock}</epgu-cf-ui-constructor-helper-text
  >`,
})
class WrapperTestComponent {}

describe('HelperTextComponent', () => {
  let component: WrapperTestComponent;
  let fixture: ComponentFixture<WrapperTestComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [WrapperTestComponent, HelperTextComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WrapperTestComponent);
    component = fixture.componentInstance;
  });

  // TODO: починить тест
  xit('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render passed content', () => {
    const helperText = fixture.debugElement.query(By.css('.helper-text')).nativeElement.textContent;
    expect(helperText).toEqual(helperTextMock);
  });
});
