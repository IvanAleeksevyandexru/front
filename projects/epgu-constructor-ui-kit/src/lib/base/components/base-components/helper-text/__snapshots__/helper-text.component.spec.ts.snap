// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperTextComponent should match snapshot 1`] = `
<ng-component>
  <epgu-cf-ui-constructor-helper-text>
    <span
      class="helper-text"
    >
      Awesome useful helper text.
    </span>
  </epgu-cf-ui-constructor-helper-text>
</ng-component>
`;
